### ISSUE TITLE
4.0b1 -Warp Mode

### ISSUE THREAD

**Issue by bluecursor** [dirkwhoffmann/virtualc64#615]
Warp mode is turned  automatically when prg is executed. Use attached prg
 
[Extreme Demo Maker PAL+NTSC[TND+MHI].D64.zip](https://github.com/dirkwhoffmann/virtualc64/files/5815093/Extreme.Demo.Maker.PAL%2BNTSC.TND%2BMHI.D64.zip)

**Comment by dirkwhoffmann** [comment]
Nice catch. Here is the thing: Warp mode is switched on, because either the C64 or the VC1541 (I didn't check) drives the IEC bus. This looks weird for the user because there is no progress indicator icon next to the drive in the status bar. This is because the drive is not rotating. Hence, it's not a bug from a technical point of view, but a bug from a users point of view (simply because it's misleading). My first thought was to ignore the disk rotation condition and show the progress indicator whenever traffic is on the IEC bus. 

On the code level this means changing this

 ```Swift
   func refreshStatusBarDriveActivity(drive: DriveID) {
        
    ...            
            if c64.iec.busy && c64.drive8.isRotating() {
                spinning8.startAnimation(self)
                spinning8.isHidden = !statusBar
            } else {
                spinning8.stopAnimation(self)
                spinning8.isHidden = true
            }
```
 to that: 
```Swift
   func refreshStatusBarDriveActivity(drive: DriveID) {
        
    ...            
            if c64.iec.busy {
                spinning8.startAnimation(self)
                spinning8.isHidden = !statusBar
            } else {
                spinning8.stopAnimation(self)
                spinning8.isHidden = true
            }
```

However, this results in this:

<img width="922" alt="Bildschirmfoto 2021-01-14 um 17 26 55" src="https://user-images.githubusercontent.com/12561945/104621255-2fabce00-5690-11eb-86a9-1c0f06589f31.png">

The problem is that the current code only knows that the IEC bus is busy, but it doesn't know the driver. There are two possibilities to solve this issue: 

-  Move the IEC traffic detection code from the IEC class to the drive class. This enables me to determine whether drive 8 or drive 9 puts traffic on the bus. Once I have this information, I could bind the progress indicator to the driving source. As a result, warp mode would still be activated for this demo, but the user would know why.

- Put tighter constraints on the warp mode enable condition. Instead of enabling warp mode whenever the IEC bus is busy, we could require the drive to be spinning as an additional constraint. As a result, warp mode wouldn't be activated for this demo.

I'll try the second solution first...

**Comment by dirkwhoffmann** [comment]
Should be fixed in v4.0b2.
